### DISCUSSION TITLE
Unable to verify and extract plaintext from signed message

### DISCUSSION THREAD

**Discussion by gladius-opensort** [krzyzanowskim/ObjectivePGP#224] [General]
Hi,

Thank you for creating an wonderful library.

I am sorry to information that I am unable to verify signature and extract text from a signed message.

The message is signed using bouncy-castle library. I am able to verify the signature using openpgpjs or keybase. 

We can reproduce the issue by verify the below signed message in https://pgptool.org/ 

Sample: PGP Signed message

-----BEGIN PGP MESSAGE-----

owEBbQGS/pANAwAKARmMxIvG/bOiAcsmYgAAAAAASGkgTWFyY2luLCB0aGlzIGEg
c2lnbmVkIG1lc3NhZ2WJATMEAAEKACcFAmRJBcIJEBmMxIvG/bOiFiEE5ysHA1Lt
xQXVwQb7GYzEi8b9s6IAAKgkB/9clRRT5HJwUOXUiHEXB2hPm03Uak4jrX+qcPqw
K0WiWFcXIkN1K4WdTzqSIRgQfTK75aQGFcffwvG67esl+sX66JtwgM/i0UKN/5zL
gL09qAiAYVEFjGKdvuXMRi8atfaU5513W7r86WXDxE+lhZKwtqyKHZl2y6Pfosm4
+yd2QKYSm+Lc9yUrxK9tVA7uPYoryJjmk7ns7lC2wdpRTVLJEfLDgzo26dCgjEkk
VeRZK3B9rDy0pOJ+32l8lIvTs8JoDXtlWznNYBtgyTehjwKHlxEpPv4conHZ9a5k
u4s6IFLF3vbpdUbfmDe5LaXfJwbFZN5/NXuzApSvBFq610ra
=WK6L
-----END PGP MESSAGE-----

My Public Key

-----BEGIN PGP PUBLIC KEY BLOCK-----

mQENBGRGLoABCADAkcrTpiNCTYJ6XbA1sNe6eSbmG/ziQx9Z5hWEgT9p6naRxKYa
xVDT4nNMy63bj1twNqjJPGIr/L8AzRzYJygXrQfLrlNWoJT1vCh1vKaIBNEMjTnY
AMnNC3rI19huTk7uCB2mrXFPGrXlCqBdbsdfF/VdTrPvZ/rg+vwgOGti4GjIga5J
cawtYwzgyfwhjsMCJX4W55dbIsR4hGMA7A+pCHxAekh/gFgsm3D8YpzoHKo346s0
zcF11oMr98ZPeqGiI2+7j0uDNfI3/9kWxWnS1c5Y44g6wMCtMIP6gmlB3+78H+SX
bj5KS0cnBiOACz7YWaK9cB59+jQLbHB/PdSHABEBAAG0H0dsYWRpdXMgMSA8Z2xh
ZGl1c0B0ZXN0aW5nLmNvbT6JAU0EEwEKAEEFAmRGLoAJEBmMxIvG/bOiFiEE5ysH
A1LtxQXVwQb7GYzEi8b9s6ICngECmwcFFgIDAQAECwkIBwUVCgkICwKZAQAAuNUI
AJ18MJ4K92kYoMAN70q32E/s44J9igjbV0IFxXAxjSxgYQxmJKKAM/vsH6tXuO8g
GVXxNX4tLxFOljsa7lKVK2xLk+1P+Sj6yWRV1hfqyjYjqid2eHAstp4jWvxW+6Bs
jDK6wJv2GDCytP1UWHi0IX9mC05BKTeU+E3k7jS7piQiSH9m5zT+VILhDGkJMrQW
d90rPpiUeuveRYz/uFFJeA62j0BJJ6rF/adBW0t+PaJA3nlXdm9mzQCP/xa1iB8x
XtJ5MwHRRoj5aaG7Q0KaEo//S83Ijwf+WpDmCJ2Wxi61lvlzRNVtIHRCfVsHXDAw
Hy2rxOlSAfBZaJr9A7XSPlU=
=8L1W
-----END PGP PUBLIC KEY BLOCK-----

<img width="1226" alt="image" src="https://user-images.githubusercontent.com/126091202/235171755-4e430b25-0440-4635-9e53-bde68dfd0f4c.png">

**Comment by krzyzanowskim** [comment]
This is due to missing support of compressed signed data. The fix will be included in next release.

**Comment by krzyzanowskim** [comment]
To extract plaintext from the signed message, you can use `decrypt`

```objc
+[ObjectivePGP decrypt:verified:certifyWithRootKey:usingKeys:passphraseForKey:decryptionError:verificationError:]`
```

then you have to check the value of verified flag and plaintext is the output

https://github.com/krzyzanowskim/ObjectivePGP/blob/6e71dd189bf47d015cb4bcaa738263e9778af4e7/Tests/PGPTests.m#L896

**Reply by kien-hoangt** [reply]
Nice @krzyzanowskim . I can get the plain text from the signed message now. Thank you very much
